modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java

```diff
@@ -206,7 +267,20 @@ public List<DropdownItem> getKBArticleDropdownItems(KBArticle kbArticle) {
 					).setActionName(
 						"/knowledge_base/delete_kb_article"
 					).setRedirect(
-						_currentURL
+						() -> {
+							if (ListUtil.isNotEmpty(selectedItemAncestorIds)) {
+								if (selectedItemAncestorIds.contains(
+										kbArticle.getResourcePrimKey())) {
+
+									return _getParentNodeURL(kbArticle);
+								}
+							}
+							else if (_isKBArticleSelected(kbArticle)) {
+								return _getParentNodeURL(kbArticle);
+							}
```

Comment:
Sorry I didn't see this in the previous review, but why not pass in `Collections.emptySet()` instead of `null` as the default value and remove the `ListUtil.isNotEmpty()` check? The same thing applies to folder down below.

Reply:
I'm passing a `null` as default value because I want to have the possibility to differentiate the case of calling the method `getKBArticleDropdownItems` without `selectedItemAncestorIds`, even if in this code I'm not treating both cases differently.

Reply:
I've finally refactored these if statements into a single one.